arguments/userSearch.js

```diff
@@ -0,0 +1,34 @@
+const { Argument, util: { regExpEsc } } = require("klasa");
+const IDRegex = /^((<@!)|())(\d{17,21})((>)|())$/;
+
+/* Possible user detection:
+ * <@!303236614623068161> (Mention)
+ * 303236614623068161 (User ID)
+ * Butterstroke (Username)
+ * Butter (Guild nickname)
+ */
+
+module.exports = class extends Argument {
+    constructor(...args) {
+        super(...args, { aliases: ["usersearch"] });
+    }
+
+    async run(arg, possible, msg) { 
+        if (arg === undefined) { return msg.author; }
+        if (IDRegex.test(arg)) { return this.client.users.get(/(\d{17,21})/.exec(arg)[0]); } 
+
+        var results = [];
+
+        if (msg.guild) {
+            var regex = new RegExp(regExpEsc(arg), "i");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Found non-literal argument to RegExp Constructor](https://app.codacy.com/manual/Butterstroke/MargarineBot/pullRequest?prid=4918293)